### PR TITLE
Supporting Stretch Edition Database in PowerShell

### PR DESCRIPTION
**Pull request by sw145**
This commit aims to supporting Stretch Edition Database in PowerShell

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@sw145__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@sw145__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+	.SYNOPSIS
+	Tests creating a stretch database on V12.0 server
+#>
+function Test-CreateStretchDatabase
+{
+	Test-CreateDatabaseInternal "12.0" "Southeast Asia"
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: We should not have hard-coded locations, as this prevents tests from running in AzureChina, AzureStack, etc.

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+function Test-UpdateStretchDatabase
+{
+	Test-UpdateDatabaseInternal "12.0" "Southeast Asia"
+}
+
+<#
+	.SYNOPSIS
+	Helper for testing updating a database
+#>
+function Test-UpdateDatabaseInternal ($serverVersion, $location = "Japan East")
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: same comment

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	}
+}
+
+
+<#
+	.SYNOPSIS
+	Tests Getting a stretch database on V12.0 server
+#>
+function Test-GetStretchDatabase
+{
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: same

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	Test-GetDatabaseInternal "12.0" "Southeast Asia"
+}
+
+<#
+	.SYNOPSIS
+	Helper for testing getting a database
+#>
+function Test-GetDatabaseInternal  ($serverVersion, $location = "Japan East")
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: Same

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+
+<#
+	.SYNOPSIS
+	Tests Deleting a stretch database on V12.0 server
+#>
+function Test-RemoveStretchDatabase
+{
+	Test-RemoveDatabaseInternal "12.0" "Southeast Asia"
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: Same

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+<#
+	.SYNOPSIS
+	Helper for testing deleting a database
+#>
+function Test-RemoveDatabaseInternal  ($serverVersion, $location = "Japan East")
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: same

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 (hunk 4 above)
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: use param to specify parameters in script

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1 (hunk 4 above)
Comment (sw145) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseCrudStretchTests.ps1]: I understand we can use existing Cmdlets to retrieve location list, but then how to select which one is the one that can satisfy every one?

Also, I just use the way how checked-in tests are testing so do i need to make this change to my code?

**Comment by sw145** [comment]
I understand we can use existing Cmdlets to retrieve location list, but then how to select which one is the one that can satisfy every one?

Also, I just use the way how checked-in tests are testing so do i need to make this change to my code?

**Comment by sw145** [comment]
I understand we can use existing Cmdlets to retrieve location list, but then how to select which one in the location list that can satisfy every developer's need? We'll still need to write scripts or change codes in the <tests.ps1> to decide locations.

Also, I just use the same way how other checked-in tests are using and i believe this is just common practice for writing tests? So do i need to really make this change to my current code?

**Comment by markcowl** [comment]
@sw145 This is the required pattern for tests.  See an example of thsi here:https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/KeyVault/Commands.KeyVault.Test/ScenarioTests/Common.ps1#L37

**Comment by sw145** [comment]
This makes more sense, i checked common.ps1 files in other resource providers and seems like not every resource provider implemented this function, so i can't find enough samples to modify my code.

The "NameSpace" seems like should be the name of the resource provider i'm currently implementing, like "Microsoft.Sql", but i don't understand what does "Type" do in the script, because each common.ps1 has diffrent values here, and i'm not sure how to follow them to modify mine.

**Comment by azurecla** [comment]
Hi __@sw145__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (shicwu). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@sw145 There are merge conflicts, please update

**Comment by sw145** [comment]
@markcowl May I know how can see these merge conflicts between my branch and Azure/dev?

**Comment by markcowl** [comment]
@sw145 if you pull the latest from the dev branch intop your branch, you will see the merge conflicts, and can fix them with  git resolve.

**Comment by sw145** [comment]
@markcowl Yeah, i know i need to pull the lastest dev into mine and compare, but the thing is i don't know how to realize this using git commands.

Since i'm in sw145/azure-powershell not in Azure/azure-powershell, and the dev branch is not up-to-date with Azure/dev, so what i'm planning to do is like following:
1. git merge from Azure/dev --> sw145/dev, to make my dev branch up-to-date
2. git merge from sw145/dev --> sw145/dev-shicwu
3. check merge conflicts, fix it, commit again
4. re-submit pullrequest
The problem right now is i can't even accomplish step #1, because i don't have authentication when try to load information about Azure/dev using command:
git fetch azure/git fetch azure-Staging

Am i doing anything wrong for the above steps? Would you like to provide an example for the right things to do with specific commands?

Thanks!

**Comment by sw145** [comment]
OK, i think i figured out the best way to do the pull, i'm using the following command:

git checkout dev-shicwu
git pull https://github.com/Azure/azure-powershell.git dev

This generates all the conflicts and i just need to solve them manually, however, what takes me most of the time is I tried to do the following:
image

**Comment by markcowl** [comment]
@sw145 The remotes you show are specific to the clone of the repo on your machine, these you set up at some point with ```git remote``` commands.  You can use either ```git pull``` or ```git rebase``` to pull in changes.  The online git docs are the best source of intformation: https://git-scm.com/documentation

**Comment by sw145** [comment]
@markcowl  sure, thanks for sharing! So is there anything i need to do for this PR to be checked-in?

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/926/

**Comment by markcowl** [comment]
LGTM once the on-demand build passes

**Comment by sw145** [comment]
@markcowl Hi, I check the on-demand build link you gave me, looks like it failed because of the wrong name for my repository, it believe we miss "s" in the name typed in the cmd.
For example:
"Cloning repository git@github.com:**w145**/azure-powershell.git"
should be 
"Cloning repository git@github.com:**sw145**/azure-powershell.git"

**Pull request merged**